contracts/treasury/TreasuryImplementation.sol

```diff
@@ -276,39 +313,138 @@ contract TreasuryImplementation is
         address _tokenAddress,
         uint256 _amountIn,
         uint256 _amountOutMin,
-        address[] memory _exchangePath,
-        uint256 _deadline
-    ) external override onlyOwner {
+        bytes memory _exchangePath
+    ) external override onlyOwnerOrImpactMarketCouncil {
         require(
             _tokenList.contains(_tokenAddress),
             "Treasury::convertAmount: this is not a valid token"
         );
 
         if (_exchangePath.length == 0) {
-            _exchangePath = _tokens[_tokenAddress].exchangePath;
+            _exchangePath = tokens[_tokenAddress].exchangePathToCUSD;
         }
 
-        if (_deadline == 0) {
-            _deadline = block.timestamp + 3600;
+        IERC20(_tokenAddress).approve(address(lpSwap.uniswapRouter()), _amountIn);
+
+        // Executes the swap.
+        uint256 _amountOut = lpSwap.uniswapRouter().exactInput(
+            IUniswapRouter02.ExactInputParams({
+                path: _exchangePath,
+                recipient: address(this),
+                amountIn: _amountIn,
+                amountOutMinimum: _amountOutMin
+            })
+        );
+
+        emit AmountConverted(_tokenAddress, _amountIn, _amountOutMin, _exchangePath, _amountOut);
+    }
+
+//    /**
+//     * @notice Transfers an amount of an ERC20 from the sender to this contract
+//     *
+//     * @param _erc20Token address of the ERC20 token
+//     * @param _amount amount of the transaction
+//     */
+//    function transferToTreasury(IERC20 _erc20Token, uint256 _amount)
+//        external
+//        override
+//        nonReentrant
+//    {
+//        _erc20Token.safeTransferFrom(msg.sender, address(this), _amount);
+//
+//        Token storage _token = tokens[address(_erc20Token)];
+//
+//        uint256 _tokenAmountToUseInLp;
+//
+//        if (_token.lpStrategy == LpStrategy.MainCoin) {
+//            _tokenAmountToUseInLp = _amount / 10;
+//        } else if (_token.lpStrategy == LpStrategy.SecondaryCoin) {
+//            _tokenAmountToUseInLp = _amount;
+//        }
+//
+//        if (_tokenAmountToUseInLp > 0) {
+//            _erc20Token.approve(address(lpSwap), _tokenAmountToUseInLp);
+//            lpSwap.addToLp(_erc20Token, _tokenAmountToUseInLp);
+//        }
+//    }
+
+    function useFundsForLP() external override onlyOwnerOrImpactMarketCouncilOrDonationMiner {
```

Comment:
I will modify it to onlyOwnerOrDonationMiner. if there are a lot of funds in the treasury, it could be a possibility to call this method  more often than once per epoch.